plugins/virtual_event/lib/model/VirtualEvent.php

```diff
@@ -97,4 +98,22 @@ public function setDeletionDueDate($v)
 	 {
 		 $this->putInCustomData(self::CUSTOM_DATA_FIELD_DELETION_DUE_DATE, $v);
 	 }
+	 
+	 public function getRegistrationFormSchema()
+	 {
+		 return $this->getFromCustomData(self::CUSTOM_DATA_FIELD_REGISTRATION_FORM_SCHEMA);
+	 }
+
+	 public function setRegistrationFormSchema($v)
+	 {
+		 // validate $v is a valid JSON object
+		 json_decode($v);
```

Comment:
I would do this validation in the validateForinsert phase and not here.
Once you do this you won't need the kVirtualEventException.php class 

also I think the check should b something lile

$json = json_decode($string);
$isValid =  (is_object($json) && json_last_error() == JSON_ERROR_NONE) ? true : false;